--- apt-0.3.19cnc52/buildlib/defaults.mak~	Wed Aug  1 23:31:58 2001
+++ apt-0.3.19cnc52/buildlib/defaults.mak	Wed Aug 15 03:09:45 2001
@@ -164,16 +164,3 @@
  endif
 endif	
 
-# Automatic -j support
-ifeq ($(NUM_PROCS),1)
-  PARALLEL_RUN=no
-endif
-
-ifndef PARALLEL_RUN
- PARALLEL_RUN=yes
- .EXPORT: PARALLEL_RUN
- # handle recursion
- ifneq ($(NUM_PROCS),)
-  MAKEFLAGS += -j $(NUM_PROCS)
- endif
-endif
