--- apt-0.3.19cnc52/buildlib/tools.m4~	Wed Aug  1 23:31:58 2001
+++ apt-0.3.19cnc52/buildlib/tools.m4	Wed Aug 15 03:38:11 2001
@@ -1,4 +1,4 @@
-AC_DEFUN(ah_HAVE_GETCONF,
+AC_DEFUN([ah_HAVE_GETCONF],
 	[AC_ARG_WITH(getconf,
 		[  --with-getconf          Enable automagical buildtime configuration],
 		[if test "$withval" = "yes"; then
@@ -14,7 +14,7 @@
 ])
 
 dnl ah_GET_CONF(variable, value ..., [default])
-AC_DEFUN(ah_GET_GETCONF,
+AC_DEFUN([ah_GET_GETCONF],
 	[AC_REQUIRE([ah_HAVE_GETCONF])
 	if test ! -z "$GETCONF";then
 		old_args="[$]@"
@@ -28,68 +28,8 @@
 		eval $1="$3"
 	fi
 ])
-AC_DEFUN(ah_NUM_CPUS,
-	[AC_MSG_CHECKING([number of cpus])
-	AC_ARG_WITH(cpus,
-		[  --with-cpus             The number of cpus to be used for building(see --with-procs, default 1)],
-		[
-		if test "$withval" = "yes"; then
-			ah_GET_GETCONF(NUM_CPUS, SC_NPROCESSORS_ONLN _NPROCESSORS_ONLN, 1)
-		elif test ! "$withval" = "no";then
-			NUM_CPUS=$withval
-		elif test "$withval" = "no";then
-			NUM_CPUS=1
-		fi],
-		[ah_GET_GETCONF(NUM_CPUS, SC_NPROCESSORS_ONLN _NPROCESSORS_ONLN, 1)]
-	)
-	ah_NUM_CPUS_msg="$NUM_CPUS"
-	if test "$NUM_CPUS" = "0"; then
-		# broken getconf, time to bitch.
-		ah_NUM_CPUS_msg="found 0 cpus.  Has someone done a lobotomy?"
-		NUM_CPUS=1
-	fi
-	if test $NUM_CPUS = 1 ;then
-		default_PROC_MULTIPLY=1
-	else
-		default_PROC_MULTIPLY=2
-	fi
-	AC_MSG_RESULT([$ah_NUM_CPUS_msg])
-	AC_SUBST(NUM_CPUS)
-])
-AC_DEFUN(ah_PROC_MULTIPLY,
-	[AC_REQUIRE([ah_NUM_CPUS])
-	AC_MSG_CHECKING([processor multiplier])
-	AC_ARG_WITH(proc-multiply,
-		[  --with-proc-multiply    Multiply this * number of cpus for parallel making(default 2).],
-		[if test "$withval" = "yes"; then
-			PROC_MULTIPLY=$default_PROC_MULTIPLY
-		elif test ! "$withval" = "no";then
-			PROC_MULTIPLY=$withval
-		fi],
-		[PROC_MULTIPLY=$default_PROC_MULTIPLY]
-	)
-	AC_MSG_RESULT([$PROC_MULTIPLY])
-	AC_SUBST(PROC_MULTIPLY)
-])
-
-AC_DEFUN(ah_NUM_PROCS,
-	[AC_REQUIRE([ah_PROC_MULTIPLY])
-	AC_REQUIRE([ah_NUM_CPUS])
-	AC_MSG_CHECKING([number of processes to run during make])
-	AC_ARG_WITH(procs,
-		[  --with-procs            The number of processes to run in parallel during make(num_cpus * multiplier).],
-		[if test "$withval" = "yes"; then
-			NUM_PROCS=`expr $NUM_CPUS \* $PROC_MULTIPLY`
-		elif test ! "$withval" = "no";then
-			NUM_PROCS=$withval
-		fi],
-		[NUM_PROCS=`expr $NUM_CPUS \* $PROC_MULTIPLY`]
-	)
-	AC_MSG_RESULT([$NUM_PROCS])
-	AC_SUBST(NUM_PROCS)
-])
 
-AC_DEFUN(rc_GLIBC_VER,
+AC_DEFUN([rc_GLIBC_VER],
 	[AC_MSG_CHECKING([glibc version])
 	dummy=if$$
 	cat <<_GLIBC_>$dummy.c
@@ -110,7 +50,7 @@
 	AC_SUBST(GLIBC_VER)
 ])
 
-AC_DEFUN(rc_LIBSTDCPP_VER,
+AC_DEFUN([rc_LIBSTDCPP_VER],
 	[AC_MSG_CHECKING([libstdc++ version])
 	dummy=if$$
 	cat <<_LIBSTDCPP_>$dummy.cc
@@ -136,7 +76,7 @@
 	AC_SUBST(LIBSTDCPP_VER)
 ])
 
-AC_DEFUN(ah_GCC3DEP,[
+AC_DEFUN([ah_GCC3DEP],[
 	AC_MSG_CHECKING(if $CXX -MD works)
 	touch gcc3dep.cc
 	${CXX-c++} -MD -o gcc3dep_test.o -c gcc3dep.cc
--- apt-0.3.19cnc52/configure.in~	Thu Aug  2 18:28:45 2001
+++ apt-0.3.19cnc52/configure.in	Wed Aug 15 03:40:32 2001
@@ -203,7 +203,6 @@
 AM_GNU_GETTEXT_VERSION([0.11.4])
 AM_GNU_GETTEXT
 
-dnl ah_NUM_PROCS
 rc_GLIBC_VER
 rc_LIBSTDCPP_VER
 dnl ah_GCC3DEP
