diff -Nur apt-0.5.4cnc7.orig/doc/apt.ent apt-0.5.4cnc7/doc/apt.ent
--- apt-0.5.4cnc7.orig/doc/apt.ent	Tue Dec 10 16:22:06 2002
+++ apt-0.5.4cnc7/doc/apt.ent	Tue Dec 10 16:44:47 2002
@@ -121,15 +121,15 @@
 
      <VarListEntry><term><option/-c/</><term><option/--config-file/</>
      <ListItem><Para>
-     Configuration File; Specify a configuration file to use. 
+     Configuration File; specify a configuration file to use. 
      The program will read the default configuration file and then this 
      configuration file. See &apt-conf; for syntax information.     
      </Para></ListItem>
      </VarListEntry>
      
      <VarListEntry><term><option/-o/</><term><option/--option/</>
      <ListItem><Para>
-     Set a Configuration Option; This will set an arbitrary configuration 
+     Set a Configuration Option; this will set an arbitrary configuration 
      option. The syntax is <option>-o Foo::Bar=bar</>.
      </Para></ListItem>
      </VarListEntry>
