--- apt-0.5.15lorg3.94a/methods/gpg.cc.orig	2008-01-12 10:45:07.000000000 +0100
+++ apt-0.5.15lorg3.94a/methods/gpg.cc	2015-02-08 16:39:02.051118648 +0100
@@ -369,7 +369,7 @@
       if (msg != NULL) 
       {
 	 removeTmpDir(TempDir, SigCount);
-	 return _error->Error(msg);
+	 return _error->Error("%s", msg);
       }
    }
    else 
@@ -393,7 +393,7 @@
 	 if (msg != NULL)
 	 {
 	    removeTmpDir(TempDir, SigCount);	       
-	    return _error->Error(msg);
+	    return _error->Error("%s", msg);
 	 }
 	 if (KeyList.empty())
 	    KeyList = KeyID;
