diff -Nur apt-0.5.4cnc7.orig/doc/apt-cdrom.8.sgml apt-0.5.4cnc7/doc/apt-cdrom.8.sgml
--- apt-0.5.4cnc7.orig/doc/apt-cdrom.8.sgml	Tue Jul 23 19:54:52 2002
+++ apt-0.5.4cnc7/doc/apt-cdrom.8.sgml	Tue Dec 10 16:43:13 2002
@@ -54,7 +54,7 @@
      <VarListEntry><Term>add</Term>
      <ListItem><Para>
      <literal/add/ is used to add a new disc to the source list. It will unmount the
-     CDROM device, prompt for a disk to be inserted and then procceed to 
+     CDROM device, prompt for a disk to be inserted and then proceed to 
      scan it and copy the index files. If the disc does not have a proper
      <filename>.disk/</> directory you will be prompted for a descriptive 
      title.
@@ -101,7 +101,7 @@
      
      <VarListEntry><term><option/-f/</><term><option/--fast/</>
      <ListItem><Para>
-     Fast Copy; Assume the package files are valid and do not check 
+     Fast Copy; assume the package files are valid and do not check 
      every package. This option should be used only if
      <command/apt-cdrom/ has been run on this disc before and did not detect 
      any errors.
@@ -110,7 +110,7 @@
      
      <VarListEntry><term><option/-a/</><term><option/--thorough/</>
      <ListItem><Para>
-     Thorough Package Scan; This option may be needed with some old 
+     Thorough Package Scan; this option may be needed with some old 
      Debian 1.1/1.2 discs that have Package files in strange places. It 
      takes much longer to scan the CD but will pick them all up.
      </VarListEntry>
@@ -120,7 +120,7 @@
                    <term><option/--recon/</>
                    <term><option/--no-act/</>
      <ListItem><Para>
-     No Changes; Do not change the &sources-list; file and do not 
+     No Changes; do not change the &sources-list; file and do not 
      write index files. Everything is still checked however.
      Configuration Item: <literal/APT::CDROM::NoAct/.
      </VarListEntry> 
diff -Nur apt-0.5.4cnc7.orig/doc/apt-get.8.sgml apt-0.5.4cnc7/doc/apt-get.8.sgml
--- apt-0.5.4cnc7.orig/doc/apt-get.8.sgml	Thu Jul 25 20:07:18 2002
+++ apt-0.5.4cnc7/doc/apt-get.8.sgml	Tue Dec 10 16:45:51 2002
@@ -270,7 +270,7 @@
      Simulate prints out
      a series of lines each one representing a dpkg operation, Configure (Conf),
      Remove (Remv), Unpack (Inst). Square brackets indicate broken packages with
-     and empty set of square brackets meaning breaks that are of no consequence
+     an empty set of square brackets meaning breaks that are of no consequence
      (rare).
      </VarListEntry>
 
@@ -286,7 +286,7 @@
 
      <VarListEntry><term><option/-u/</><term><option/--show-upgraded/</>
      <ListItem><Para>
-     Show upgraded packages; Print out a list of all packages that are to be
+     Show upgraded packages; print out a list of all packages that are to be
      upgraded. 
      Configuration Item: <literal/APT::Get::Show-Upgraded/.
      </VarListEntry>
@@ -300,7 +300,7 @@
 
      <VarListEntry><term><option/--ignore-hold/</>
      <ListItem><Para>
-     Ignore package Holds; This causes <command/apt-get/ to ignore a hold 
+     Ignore package Holds; this causes <command/apt-get/ to ignore a hold 
      placed on a package. This may be useful in conjunction with 
      <literal/dist-upgrade/ to override a large number of undesired holds. 
      Configuration Item: <literal/APT::Ignore-Hold/.
@@ -308,7 +308,7 @@
 
      <VarListEntry><term><option/--no-upgrade/</>
      <ListItem><Para>
-     Do not upgrade packages; When used in conjunction with <literal/install/
+     Do not upgrade packages; when used in conjunction with <literal/install/
      <literal/no-upgrade/ will prevent packages listed from being upgraded
      if they are already installed. 
      Configuration Item: <literal/APT::Get::Upgrade/.
@@ -316,7 +316,7 @@
 
      <VarListEntry><term><option/--force-yes/</>
      <ListItem><Para>
-     Force yes; This is a dangerous option that will cause apt to continue 
+     Force yes; this is a dangerous option that will cause apt to continue 
      without prompting if it is doing something potentially harmful. It 
      should not be used except in very special situations. Using 
      <literal/force-yes/ can potentially destroy your system! 
diff -Nur apt-0.5.4cnc7.orig/doc/apt.conf.5.sgml apt-0.5.4cnc7/doc/apt.conf.5.sgml
--- apt-0.5.4cnc7.orig/doc/apt.conf.5.sgml	Tue Dec 10 16:22:06 2002
+++ apt-0.5.4cnc7/doc/apt.conf.5.sgml	Tue Dec 10 16:47:05 2002
@@ -76,7 +76,7 @@
 
  <RefSect1><Title>The APT Group</>
    <para>   
-   This group of options controls general APT behavior as well as holding the
+   This group of options controls general APT behaviour as well as holding the
    options for all of the tools.
 
    <VariableList>
@@ -89,7 +89,7 @@
      
      <VarListEntry><Term>Ignore-Hold</Term>
      <ListItem><Para>
-     Ignore Held packages; This global option causes the problem resolver to
+     Ignore Held packages; this global option causes the problem resolver to
      ignore held packages in its decision making. 
      </VarListEntry>
 
@@ -103,7 +103,7 @@
 
      <VarListEntry><Term>Immediate-Configure</Term>
      <ListItem><Para>
-     Disable Immediate Configuration; This dangerous option disables some
+     Disable Immediate Configuration; this dangerous option disables some
      of APT's ordering code to cause it to make fewer dpkg calls. Doing
      so may be necessary on some extremely slow single user systems but 
      is very dangerous and may cause package install scripts to fail or worse.
diff -Nur apt-0.5.4cnc7.orig/doc/apt.ent apt-0.5.4cnc7/doc/apt.ent
--- apt-0.5.4cnc7.orig/doc/apt.ent	Tue Dec 10 16:22:06 2002
+++ apt-0.5.4cnc7/doc/apt.ent	Tue Dec 10 16:44:47 2002
@@ -121,15 +121,15 @@
 
      <VarListEntry><term><option/-c/</><term><option/--config-file/</>
      <ListItem><Para>
-     Configuration File; Specify a configuration file to use. 
+     Configuration File; specify a configuration file to use. 
      The program will read the default configuration file and then this 
      configuration file. See &apt-conf; for syntax information.     
      </Para></ListItem>
      </VarListEntry>
      
      <VarListEntry><term><option/-o/</><term><option/--option/</>
      <ListItem><Para>
-     Set a Configuration Option; This will set an arbitrary configuration 
+     Set a Configuration Option; this will set an arbitrary configuration 
      option. The syntax is <option>-o Foo::Bar=bar</>.
      </Para></ListItem>
      </VarListEntry>
