--- apt-0.5.5cnc4.1/apt-pkg/acquire-item.cc.orig	2003-03-08 00:11:34.000000000 +0100
+++ apt-0.5.5cnc4.1/apt-pkg/acquire-item.cc	2003-06-19 22:59:35.000000000 +0200
@@ -27,6 +27,7 @@
 #include <apt-pkg/repository.h>
 #include <apt-pkg/md5.h>
 #include <iostream>
+#include <cassert>
 using namespace std;
 
 #include <apti18n.h>
--- apt-0.5.5cnc4.1/apt-pkg/rpm/rpmsrcrecords.cc.orig	2003-03-08 00:12:41.000000000 +0100
+++ apt-0.5.5cnc4.1/apt-pkg/rpm/rpmsrcrecords.cc	2003-06-19 23:01:29.000000000 +0200
@@ -30,6 +30,8 @@
 #include <rpm/rpmds.h>
 #endif
 
+#include <cassert>
+
 // SrcRecordParser::rpmSrcRecordParser - Constructor			/*{{{*/
 // ---------------------------------------------------------------------
 /* */
--- apt-0.5.5cnc4.1/apt-pkg/rpm/rpmversion.cc.orig	2003-03-08 00:12:34.000000000 +0100
+++ apt-0.5.5cnc4.1/apt-pkg/rpm/rpmversion.cc	2003-06-19 23:02:00.000000000 +0200
@@ -31,6 +31,8 @@
 #include <rpm/rpmds.h>
 #endif
 
+#include <cassert>
+
 rpmVersioningSystem rpmVS;
 
 // rpmVS::rpmVersioningSystem - Constructor				/*{{{*/
