diff -Nur apt-0.5.4cnc7.orig/doc/apt-get.8 apt-0.5.4cnc7/doc/apt-get.8
--- apt-0.5.4cnc7.orig/doc/apt-get.8	Tue Jul 23 19:54:52 2002
+++ apt-0.5.4cnc7/doc/apt-get.8	Tue Dec 10 16:08:46 2002
@@ -384,7 +384,7 @@
 .SH "SEE ALSO"
 .PP
 \fBapt-cache\fR(8), \fBapt-cdrom\fR(8), \fBdpkg\fR(8), \fBdselect\fR(8), \fB\fIsources.list\fB\fR(5), \fB\fIapt.conf\fB\fR(5), 
-The APT users guide in /usr/share/doc/apt/, \fBapt_preferences\fR(5).
+The APT users guide in /usr/share/doc/apt*/, \fBapt_preferences\fR(5).
 .SH "DIAGNOSTICS"
 .PP
 \fBapt-get\fR returns zero on normal operation, decimal 100 on error.
diff -Nur apt-0.5.4cnc7.orig/doc/apt.conf.5 apt-0.5.4cnc7/doc/apt.conf.5
--- apt-0.5.4cnc7.orig/doc/apt.conf.5	Tue Jul 23 19:54:52 2002
+++ apt-0.5.4cnc7/doc/apt.conf.5	Tue Dec 10 16:16:50 2002
@@ -58,7 +58,7 @@
 .fi
 .PP
 In general the sample configuration file in 
-\fI/usr/share/doc/apt//examples/apt.conf\fR \fI/usr/share/doc/apt/examples/configure-index.gz\fR
+\fI/usr/share/doc/apt*/examples/apt.conf\fR \fI/usr/share/doc/apt*/examples/configure-index.gz\fR
 is a good guide for how it should look.
 .PP
 Two specials are allowed, #include and #clear. 
@@ -178,7 +178,7 @@
 proxy you will have to set the ftp::ProxyLogin script in the 
 configuration file. This entry specifies the commands to send to tell 
 the proxy server what to connect to. Please see 
-\fI/usr/share/doc/apt/examples/configure-index.gz\fR for an example of 
+\fI/usr/share/doc/apt*/examples/configure-index.gz\fR for an example of 
 how to do this. The subsitution variables available are 
 $(PROXY_USER), $(PROXY_PASS), $(SITE_USER),
 $(SITE_PASS), $(SITE), and $(SITE_PORT).
@@ -329,7 +329,7 @@
 disable the inclusion of statfs data in CDROM IDs.
 .SH "EXAMPLES"
 .PP
-\fI/usr/share/doc/apt/examples/configure-index.gz\fR contains a 
+\fI/usr/share/doc/apt*/examples/configure-index.gz\fR contains a 
 sample configuration file showing the default values for all possible 
 options.
 .SH "FILES"
diff -Nur apt-0.5.4cnc7.orig/doc/apt.conf.5.sgml apt-0.5.4cnc7/doc/apt.conf.5.sgml
--- apt-0.5.4cnc7.orig/doc/apt.conf.5.sgml	Thu Jul 25 20:07:18 2002
+++ apt-0.5.4cnc7/doc/apt.conf.5.sgml	Tue Dec 10 16:19:21 2002
@@ -59,7 +59,7 @@
 </programlisting></informalexample>
    <para>
    In general the sample configuration file in 
-   <filename>&docdir;/examples/apt.conf</> &configureindex;
+   <filename>&docdir;examples/apt.conf</> &configureindex;
    is a good guide for how it should look.
    <para>
    Two specials are allowed, <literal/#include/ and <literal/#clear/. 
diff -Nur apt-0.5.4cnc7.orig/doc/apt.ent apt-0.5.4cnc7/doc/apt.ent
--- apt-0.5.4cnc7.orig/doc/apt.ent	Thu Jul 25 20:07:18 2002
+++ apt-0.5.4cnc7/doc/apt.ent	Tue Dec 10 16:09:18 2002
@@ -1,7 +1,7 @@
 <!-- -*- mode: sgml; mode: fold -*- -->
 
 <!-- Some common paths.. -->
-<!ENTITY docdir "/usr/share/doc/apt/">
+<!ENTITY docdir "/usr/share/doc/apt*/">
 <!ENTITY configureindex "<filename>&docdir;examples/configure-index.gz</>">
 <!ENTITY aptconfdir "<filename>/etc/apt.conf</>">
 <!ENTITY statedir "/var/lib/apt">
diff -Nur apt-0.5.4cnc7.orig/doc/fr/apt.ent.fr apt-0.5.4cnc7/doc/fr/apt.ent.fr
--- apt-0.5.4cnc7.orig/doc/fr/apt.ent.fr	Tue Jul 23 19:54:52 2002
+++ apt-0.5.4cnc7/doc/fr/apt.ent.fr	Tue Dec 10 16:21:02 2002
@@ -1,7 +1,7 @@
 <!-- -*- mode: sgml; mode: fold -*- -->
 <!-- translation of version 1.7 -->
 <!-- Some common paths.. -->
-<!ENTITY docdir "/usr/share/doc/apt/">
+<!ENTITY docdir "/usr/share/doc/apt*/">
 <!ENTITY configureindex "<filename>&docdir;examples/configure-index.gz</>">
 <!ENTITY aptconfdir "<filename>/etc/apt.conf</>">
 <!ENTITY statedir "/var/lib/apt">
diff -Nur apt-0.5.4cnc7.orig/doc/makefile apt-0.5.4cnc7/doc/makefile
--- apt-0.5.4cnc7.orig/doc/makefile	Tue Jul 23 19:54:52 2002
+++ apt-0.5.4cnc7/doc/makefile	Tue Dec 10 16:20:41 2002
@@ -1,6 +1,6 @@
 # -*- make -*-
 BASE=..
-SUBDIR=doc
+SUBDIR=doc doc/pl
 
 # Bring in the default rules
 include ../buildlib/defaults.mak
@@ -13,7 +13,8 @@
 # Man pages
 SOURCE = apt-cache.8 apt-get.8 apt-cdrom.8 apt.conf.5 sources.list.5 \
          apt-config.8 apt-sortpkgs.1 apt-ftparchive.1 apt_preferences.5 \
-	 apt-extracttemplates.1 vendors.list.5
+	 apt-extracttemplates.1 vendors.list.5 \
+	 pl/apt-cache.8 pl/apt-get.8 pl/apt-cdrom.8
 INCLUDES = apt.ent	 
 include $(SGML_MANPAGE_H)
 
