diff -Nur apt-0.5.4cnc7.orig/doc/apt.ent apt-0.5.4cnc7/doc/apt.ent
--- apt-0.5.4cnc7.orig/doc/apt.ent	Thu Jul 25 20:07:18 2002
+++ apt-0.5.4cnc7/doc/apt.ent	Tue Dec 10 16:09:18 2002
@@ -1,7 +1,7 @@
 <!-- -*- mode: sgml; mode: fold -*- -->
 
 <!-- Some common paths.. -->
-<!ENTITY docdir "/usr/share/doc/apt/">
+<!ENTITY docdir "/usr/share/doc/apt*/">
 <!ENTITY configureindex "<filename>&docdir;examples/configure-index.gz</>">
 <!ENTITY aptconfdir "<filename>/etc/apt.conf</>">
 <!ENTITY statedir "/var/lib/apt">
