diff -urN apt-0.3.19cnc38.org/apt-pkg/rpm/rpmlistparser.cc apt-0.3.19cnc38/apt-pkg/rpm/rpmlistparser.cc
--- apt-0.3.19cnc38.org/apt-pkg/rpm/rpmlistparser.cc	Thu Apr 19 00:38:38 2001
+++ apt-0.3.19cnc38/apt-pkg/rpm/rpmlistparser.cc	Thu Apr 19 00:38:54 2001
@@ -348,7 +348,6 @@
 unsigned short rpmListParser::VersionHash()
 {
    int Sections[] ={
-      RPMTAG_REQUIRENAME,
       RPMTAG_OBSOLETENAME,
       RPMTAG_CONFLICTNAME,
 	  0
