diff -Nur apt-0.5.4cnc7.orig/doc/apt.conf.5.sgml apt-0.5.4cnc7/doc/apt.conf.5.sgml
--- apt-0.5.4cnc7.orig/doc/apt.conf.5.sgml	Thu Jul 25 20:07:18 2002
+++ apt-0.5.4cnc7/doc/apt.conf.5.sgml	Tue Dec 10 16:19:21 2002
@@ -59,7 +59,7 @@
 </programlisting></informalexample>
    <para>
    In general the sample configuration file in 
-   <filename>&docdir;/examples/apt.conf</> &configureindex;
+   <filename>&docdir;examples/apt.conf</> &configureindex;
    is a good guide for how it should look.
    <para>
    Two specials are allowed, <literal/#include/ and <literal/#clear/. 
diff -Nur apt-0.5.4cnc7.orig/doc/apt.ent apt-0.5.4cnc7/doc/apt.ent
--- apt-0.5.4cnc7.orig/doc/apt.ent	Thu Jul 25 20:07:18 2002
+++ apt-0.5.4cnc7/doc/apt.ent	Tue Dec 10 16:09:18 2002
@@ -1,7 +1,7 @@
 <!-- -*- mode: sgml; mode: fold -*- -->
 
 <!-- Some common paths.. -->
-<!ENTITY docdir "/usr/share/doc/apt/">
+<!ENTITY docdir "/usr/share/doc/apt*/">
 <!ENTITY configureindex "<filename>&docdir;examples/configure-index.gz</>">
 <!ENTITY aptconfdir "<filename>/etc/apt.conf</>">
 <!ENTITY statedir "/var/lib/apt">
diff -Nur apt-0.5.4cnc7.orig/doc/fr/apt.ent.fr apt-0.5.4cnc7/doc/fr/apt.ent.fr
--- apt-0.5.4cnc7.orig/doc/fr/apt.ent.fr	Tue Jul 23 19:54:52 2002
+++ apt-0.5.4cnc7/doc/fr/apt.ent.fr	Tue Dec 10 16:21:02 2002
@@ -1,7 +1,7 @@
 <!-- -*- mode: sgml; mode: fold -*- -->
 <!-- translation of version 1.7 -->
 <!-- Some common paths.. -->
-<!ENTITY docdir "/usr/share/doc/apt/">
+<!ENTITY docdir "/usr/share/doc/apt*/">
 <!ENTITY configureindex "<filename>&docdir;examples/configure-index.gz</>">
 <!ENTITY aptconfdir "<filename>/etc/apt.conf</>">
 <!ENTITY statedir "/var/lib/apt">
 
